### PR TITLE
add SX126X radio option and update driver lib

### PR DESCRIPTION
**Pull request by bentcooke**
This PR updates the example to use the latest driver lib that includes SX126X support and adds the necessary options to use this radio.  A couple of examples targets are added as well to show the pinout differences from the existing sx1272,sx1276 radios.

### CODE REVIEW

**Comment by bentcooke** [comment]
@hasnainvirk @AnttiKauppila

File: mbed_app.json [hunk 1]
```diff
@@ -35,28 +40,40 @@
...
+            "lora-dio1":           "D5",
+            "lora-busy":           "D3",
+            "lora-freq-sel":       "A1",
+            "lora-dev-sel":        "A2",
+            "lora-xtal-sel":       "A3",
+            "lora-ant-switch":     "D8"
         },
 
         "K64F": {
+            "lora-radio":          "SX126X",
```
Comment (AnttiKauppila) [inline mbed_app.json]: Default should not be overridden here

File: mbed_app.json [hunk 2]
```diff
@@ -35,28 +40,40 @@
             "lora.phy": "EU868",
             "lora.device-eui": "{ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 }",
             "lora.application-eui": "{ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 }",
-            "lora.application-key": "{ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 }"
+            "lora.application-key": "{ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 }"        
+        },
+
+        "NUCLEO_L073RZ": {
+            "lora-radio":          "SX126X",
```
Comment (AnttiKauppila) [inline mbed_app.json]: Default should not be overridden here

File: mbed_app.json [hunk 3]
```diff
@@ -35,28 +40,40 @@
...
+            "SX126X-lora-driver.spi-frequency": 12000000,
```
Comment (AnttiKauppila) [inline mbed_app.json]: This override should not be part of the example app

File: mbed_app.json [hunk 4]
```diff
@@ -35,28 +40,40 @@
...
+            "lora-busy":           "D3",
+            "lora-freq-sel":       "A1",
+            "lora-dev-sel":        "A2",
+            "lora-xtal-sel":       "A3",
+            "lora-ant-switch":     "D8"
         },
 
         "K64F": {
+            "lora-radio":          "SX126X",
+            "SX126X-lora-driver.spi-frequency": 12000000,
```
Comment (AnttiKauppila) [inline mbed_app.json]: This override should not be part of the example app

**Comment by henri98** [comment]
Any updates on this?

**Comment by bentcooke** [comment]
Also, the forced push came from rebasing all the changes onto current master.

**Comment by AnttiKauppila** [comment]
@bentcooke You haven't fixed my proposed changes.

**Comment by bentcooke** [comment]
@AnttiKauppila I believe your comments are addressed now regarding the default settings in mbed_app.json.

**Pull request merged**